packages/aws-cdk-lib/aws-ec2/test/instance.test.ts

```diff
@@ -904,4 +904,73 @@ test('specify ebs optimized instance', () => {
     InstanceType: 't3.large',
     EbsOptimized: true,
   });
-});
\ No newline at end of file
+});
+
+test.each([
+  [true, true],
+  [false, false],
+])('given nitroEnclaveEnabled %p', (given: boolean, expected: boolean) => {
+  // WHEN
+  new Instance(stack, 'Instance', {
+    vpc,
+    machineImage: new AmazonLinuxImage(),
+    instanceType: InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE),
+    nitroEnclaveEnabled: given,
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::EC2::Instance', {
+    EnclaveOptions: {
+      Enabled: expected,
+    },
+  });
+});
+
+test.each([
+  [true, true],
+  [false, false],
+])('given hibernationConfigured %p', (given: boolean, expected: boolean) => {
+  // WHEN
+  new Instance(stack, 'Instance', {
+    vpc,
+    machineImage: new AmazonLinuxImage(),
+    instanceType: InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE),
+    hibernationConfigured: given,
+    blockDevices: [{
+      deviceName: '/dev/xvda',
+      volume: BlockDeviceVolume.ebs(30, {
+        volumeType: EbsDeviceVolumeType.GP3,
+        encrypted: true,
+        deleteOnTermination: true,
+      }),
+    }],
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::EC2::Instance', {
+    HibernationOptions: {
+      Configured: expected,
+    },
+  });
+});
+
+test('throw if AWS Nitro Enclaves and hibernation are enabled', () => {
```

Comment:
It would be good to use parameter names in the sentence.

Reply:
I changed the sentence to match parameter names.